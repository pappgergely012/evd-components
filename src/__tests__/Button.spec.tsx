import React from "react";
import { Button } from "../index";
import renderer, { act } from "react-test-renderer";


const component = renderer.create(
  <Button
    title="Order"
    className="primary"
    onClick={() => console.log('Button clicked!')}
    type="square"
    position="center"
    style={{backgroundColor: "#fff", color: "#333", border: "1px solid #fff"}}
  />
);

test("Button", () => {
  const instance   = component.root;
  const inst: any  = component.toJSON();

  expect(instance.props.className).toBe("primary");
  expect(instance.props.title).toBe("Order");
  expect(instance.props.type).toBe("square");
  expect(instance.props.position).toBe("center");

  act(() =>Â {
    inst.props.onClick();
  })

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});