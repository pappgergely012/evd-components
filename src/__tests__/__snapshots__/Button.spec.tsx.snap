// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  className="button square primary center"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "border": "1px solid #fff",
      "color": "#333",
    }
  }
  type="button"
>
  Order
</button>
`;
